src/main/java/Duke.java

```diff
@@ -1,10 +1,124 @@
+        System.out.println(input.taskStatusWithDescriptionText());
+        System.out.println("Now you have " + taskSize + " tasks in the list.");
+        printHorizontalLine();
+    }
+
+    public static void printTaskList() {
+        if (taskSize == 0) {
+            System.out.println("There is nothing on your list!");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 1; i < taskSize + 1; i++) {
+                System.out.println(i + "." + taskList[i].taskStatusWithDescriptionText());
+            }
+        }
+        printHorizontalLine();
+    }
+
+    public static void printMarkTaskText(int taskNumber, String command, boolean isDone) {
+        String previousIcon = taskList[taskNumber].getStatusIcon();
+        if (previousIcon == "X") {
+            if (isDone) {
+                System.out.println("This task has already been marked!");
+            } else {
+                System.out.println("OK, I've marked this task as not done yet:");
+            }
+        } else {
+            if (!isDone) {
+                System.out.println("This task has already been unmarked!");
+            } else {
+                System.out.println("Nice! I've marked this task as done:");
+            }
+        }
+        taskList[taskNumber].setDone(isDone);
+        String newIcon = taskList[taskNumber].getStatusIcon();
+        System.out.println("[" + newIcon + "] " + taskList[taskNumber].description);
+        printHorizontalLine();
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printGreeting();
+        String input;
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            input = in.nextLine();
+            String[] inputWords = input.split(" ", 2);
+            switch (inputWords[0]) {
+            case "bye":
+                printExitText();
+                break;
+            case "list":
+                printTaskList();
+                break;
+            case "mark":
+                int taskNumber =  Integer.parseInt(inputWords[1]);
+                printMarkTaskText(taskNumber, "mark", true);
+                break;
+            case "unmark":
+                taskNumber =  Integer.parseInt(inputWords[1]);
+                printMarkTaskText(taskNumber, "unmark", false);
+                break;
+            case "todo":
+                Todo newTodo = new Todo(inputWords[1], 'T');
+                addItem(newTodo);
+                printAddItemText(newTodo);
+                break;
+            case "deadline":
+                String[] taskWithDeadline = inputWords[1].split("/by ", 2);
```

Comment:
Could 2 be a magic number here?

https://nus-cs2113-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers